### ISSUE TITLE
port the extension to Firefox

### ISSUE THREAD

**Issue by doshidak** [doshidak/showdex#4]
> **Requested by:**
> * [**lighthouse64**](https://www.smogon.com/forums/threads/showdex-an-auto-updating-damage-calculator-built-into-showdown.3707265/post-9324048)
> * [**mdragon13**](https://www.reddit.com/r/pokemonshowdown/comments/x5bi27/showdex_an_autoupdating_damage_calculator_built/in0zpcd/)
> * [**kirito_1707**](https://www.reddit.com/r/pokemonshowdown/comments/x5bi27/showdex_an_autoupdating_damage_calculator_built/in7624p/)

Mozilla offers a pretty cool tool called [extensiontest.com](https://extensiontest.com), which determines extension compatibility with Firefox from the generated CRX file.

Based on just uploading the CRX that was provided to the Chrome Web Store, seems that it's compatible as-is:

![Image](https://user-images.githubusercontent.com/18254287/188297350-d459bade-70ec-4982-9993-6c90d10f03b0.png)

According to the report, this just needs some minor tweaking for the extension ID in the `manifest.json`:

```json
{
  "compat": [],
  "errors": [
    {
      "message": "\"/manifest_version\" must be <= 2",
      "description": "Your JSON file could not be parsed.",
      "locations": [
        {}
      ]
    },
    {
      "message": "The extension ID is required in Manifest Version 3 and above.",
      "description": "See https://mzl.la/3PLZYdo for more information.",
      "locations": [
        {
          "file": "manifest.json"
        }
      ]
    },
    {
      "message": "File is too large to parse.",
      "description": "This file is not binary and is too large to parse. Files larger than 4MB will not be parsed. Consider moving large lists of data out of JavaScript files and into JSON files, or splitting very large files into smaller ones.",
      "locations": [
        {
          "file": "main.js"
        }
      ]
    }
  ],
  "warnings": [
    {
      "message": "\"runtime.onMessageExternal\" can cause issues when loaded temporarily",
      "description": "This API can cause issues when loaded temporarily using about:debugging in Firefox unless you specify applications|browser_specific_settings > gecko > id in the manifest. Please see: https://mzl.la/2hizK4a for more.",
      "locations": [
        {
          "file": "background.js",
          "line": 1
        }
      ]
    }
  ]
}
```

**Comment by doshidak** [comment]
Initially thought this would be a walk in the park, but turns out Firefox doesn't support MV3 (Manifest v3) out-of-the-box, and even if MV3 was enabled, manifest properties like `externally_connectable` [aren't supported yet](https://bugzilla.mozilla.org/show_bug.cgi?id=1319168).

Thus, I had to create a *separate* manifest for Firefox, using MV2 and tweak some of the code to work with the `browser` global (from the WebExtensions API), instead of `chrome`.

Long story short, I got it working on Firefox. Yay!

![firefox-screenshot](https://user-images.githubusercontent.com/18254287/188522466-b6709411-9b47-49a7-8476-07dffb0cf35a.png)

Will close this via reference on the next PR.
